### ISSUE TITLE
New properties in core schemas

### ISSUE THREAD

**Issue by romanrostislavovich** [linterhub/schema#95]
Please, add following properties to `deps`, `package` and `linter` schemas:
- `$version` - describe a version of the schema
- `$schema` - used to declare that a JSON fragment is actually a piece of JSON Schema

**Comment by xferra** [comment]
Sorry, this is part of the expected result. Let me rephrase: all schemas based on schemaver should have those optional attributes (we may even make them required).
Versioned JSON objects should support it as well. Here is my staged draft:
```json
{
    "$schema": "https://schema.linterhub.com/schemaver.json",
    "$id": "https://schema.linterhub.com/schemaver.json",
    "$version": "0.0.0",
    "title": "Versioned Schema",
    "description": "The meta-schema of versioned JSON schema",
    "type": "object",
    "$ref": "#/definitions/schemaver",
    "definitions": {
        "property": {
            "title": "Version",
            "description": "The value of object version",
            "type": "string",
            "pattern": "^[0-9]+.[0-9]+.[0-9]+$"
        },
        "object": {
            "title": "Versioned Object",
            "description": "The versioned object",
            "type": "object",
            "properties": {
                "$version": {
                    "title": "Version",
                    "description": "The version value",
                    "$ref": "#/definitions/property"
                }
            },
            "required": [
                "$version"
            ]
        },
        "schema": {
            "title": "JSON Schema",
            "description": "The meta-schema of JSON schema",
            "type": "object",
            "$ref": "http://json-schema.org/draft-06/schema#"
        },
        "schemaver": {
            "title": "Versioned Schema",
            "description": "The meta-schema of versioned JSON schema",
            "type": "object",
            "allOf": [
                {
                    "$ref": "#/definitions/schema"
                },
                {
                    "$ref": "#/definitions/object"
                }
            ]
        }
    }
}
```

PS: Only partially implemented, but hope it helps.

**Comment by xferra** [comment]
How to use it:

Let's say we define another versioned schema which describes a versioned object (maybe move it to `objectver.json` or smth similar is a good idea - ref above is too long):
```json
{
    "$schema": "https://schema.linterhub.com/schemaver.json",
    "$version": "1.0.0",
    "allOf": [
        { "$ref": "https://schema.linterhub.com/schemaver.json#/definitions/object" },
        { "$ref": "#/definitions/type" }
    ],
    "definitions": {
        "type": {
            "properties": {
                "prop": {
                    "type": "string"
                }
            }
        }
    }
}
```
Then the corresponding JSON data may looks like:
```json
{
    "$schema": "my://best.shema/ever.json",
    "$version": "1.0.0",
    "prop": "value"
}
```
